x-pack/plugins/security_solution/cypress/e2e/detection_rules/persistent_rules_table_state.cy.ts

```diff
@@ -0,0 +1,356 @@
+  expectRowsPerPage(20);
+  expectTablePage(1);
+}
+
+function expectManagementTableRules(ruleNames: string[]): void {
+  expectNumberOfRules(RULES_MANAGEMENT_TABLE, ruleNames.length);
+
+  for (const ruleName of ruleNames) {
+    expectToContainRule(RULES_MANAGEMENT_TABLE, ruleName);
+  }
+}
+
+describe('Persistent rules table state', () => {
+  before(() => {
+    cleanKibana();
+    createTestRules();
+    login();
+  });
+
+  beforeEach(() => {
+    resetRulesTableState();
+  });
+
+  describe('while on a happy path', () => {
+    it('activates management tab by default', () => {
+      visit(SECURITY_DETECTIONS_RULES_URL);
+
+      expectRulesManagementTab();
+    });
+
+    it('prefers url state over storage state', () => {
+      setStorageState({
+        searchTerm: 'test',
+        tags: ['tag-c'],
+        source: 'prebuilt',
+        enabled: true,
+        field: 'severity',
+        order: 'desc',
+        perPage: 10,
+      });
+
+      visitRulesTableWithState({
+        searchTerm: 'rule',
+        tags: ['tag-b'],
+        source: 'custom',
+        enabled: false,
+        field: 'name',
+        order: 'asc',
+        perPage: 5,
+        page: 2,
+      });
+
+      expectRulesManagementTab();
+      expectRulesTableState();
+      expectTablePage(2);
+      expectManagementTableRules(['rule 6']);
+    });
+
+    describe('and on the rules management tab', () => {
+      beforeEach(() => {
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+      });
+
+      it('persists after reloading the page', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        cy.reload();
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(2);
+      });
+
+      it('persists after navigating back from a rule details page', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        goToRuleDetails();
+        cy.go('back');
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(2);
+      });
+
+      it('persists after navigating to another page inside Security Solution', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        visit(DASHBOARDS_URL);
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(1);
+      });
+
+      it('persists after navigating to another page outside Security Solution', () => {
+        changeRulesTableState();
+        goToTablePage(2);
+
+        visit(KIBANA_HOME);
+        visit(SECURITY_DETECTIONS_RULES_MANAGEMENT_URL);
+
+        expectRulesManagementTab();
+        expectRulesTableState();
+        expectTablePage(1);
+      });
+    });
+
+    describe('and on the rules monitoring tab', () => {
+      beforeEach(() => {
+        visit(SECURITY_DETECTIONS_RULES_MONITORING_URL);
+
+        changeRulesTableState();
+        goToTablePage(2);
```

Comment:
Same as https://github.com/elastic/kibana/pull/149638/files#r1106910760, let's move these steps to test bodies. 

And do we need to change the page number to test that the correct tab is selected?